.github/scripts/create-flaky-test-report.ts

```diff
@@ -0,0 +1,148 @@
+        );
+        if (!firefoxTestReport) return [];
+
+        return [chromeTestReport, firefoxTestReport];
+      }),
+    )
+  ).flat();
+
+  const testRuns: TestRun[] = (
+    await Promise.all(
+      artifacts.map(async (artifact) => {
+        const response = await github.rest.actions.downloadArtifact({
+          owner: env.OWNER,
+          repo: env.REPOSITORY,
+          artifact_id: artifact.id,
+          archive_format: 'zip',
+        });
+        const zip = await JSZip.loadAsync(response.data as ArrayBuffer);
+        const file = zip.files['test-runs.json'];
+        if (!file) throw new Error(`'test-runs.json' file in zip not found!`);
+        const content = await file.async('string');
+        return JSON.parse(content);
+      }),
+    )
+  ).flat();
+
+  const failedTests = testRuns.flatMap((testRun) =>
+    testRun.testFiles.flatMap((testFile) =>
+      testFile.testSuites.flatMap((testSuite) =>
+        testSuite.testCases
+          .filter((testCase) => testCase.status === 'failed')
+          .map((testCase) => ({ ...testCase, path: testFile.path })),
+      ),
+    ),
+  );
+
+  const summarizedFailedTests = Object.values(
+    failedTests.reduce<
+      Record<string, (typeof failedTests)[number] & { count: number }>
+    >((summary, test) => {
+      if (summary[test.name]) {
+        summary[test.name].count += 1;
+      } else {
+        summary[test.name] = { ...test, count: 1 };
+      }
+      return summary;
+    }, {}),
+  ).sort((a, b) => b.count - a.count);
+
+  const options = { year: 'numeric', month: 'long', day: 'numeric' } as const;
+  const fromDateString = from.toLocaleDateString('en-US', options);
+  const toDateString = to.toLocaleDateString('en-US', options);
+
+  console.log(
+    `❌ Failed tests on the ${env.REPOSITORY} repository ${env.BRANCH} branch from ${fromDateString} to ${toDateString}`,
+  );
+  const lines = [
+    `\n\n:x: Failed tests on the <https://github.com/${env.OWNER}/${env.REPOSITORY}|${env.REPOSITORY}> repository <https://github.com/${env.OWNER}/${env.REPOSITORY}/tree/${env.BRANCH}|${env.BRANCH}> branch from ${fromDateString} to ${toDateString}`,
+  ];
+
+  if (!summarizedFailedTests.length) {
+    console.log('No failed tests found, good job!');
+    lines.push('\n No failed tests found, good job!');
+  } else {
+    for (const test of summarizedFailedTests) {
+      const issue = {
+        template: 'general-issue.yml',
+        labels: ['type-bug', 'Sev2-normal', 'flaky tests'].toString(),
+        title: `Flaky test: \`${test.name}\``,
+        description: `\`\`\`js\n${test.error}\n\`\`\``,
+      };
+      const params = new URLSearchParams(issue);
+      console.error(
+        `• ${test.name} failed ${test.count} time${test.count > 1 ? 's' : ''}`,
+      );
+      lines.push(
+        `\n• <https://github.com/${env.OWNER}/${env.REPOSITORY}/blob/${
+          env.BRANCH
+        }/${test.path}|${test.name}> failed ${test.count} time${
```

Comment:
I think you'll need to escape these for use in a URL. You can probably use `new URL()` and build the path using

```
url.pathname = `/blob/${env.BRANCH}/${test.path}|${test.name}`
// and
url.toString();
```

so that if any variables, like `test.name` (or other variables) contains a `">"`, it doesn't escape the opening `<` link, and instead will output `%3E` in places of `">"`.

Reply:
I think you URLs are still broken, if `test.name` or `test.path` have special URL chars, they'll be treated as URL chars, and not properly escaped. You'll need to escape them yourself or set them via `url.pathname = path`, which will escape them for you.